### PR TITLE
Adjust sensor to better support large jira projects

### LINKED ISSUE #45: Sensor times out searching for issues with very large JIRA project
I have a JIRA project which has ~ 35k issues. I noticed that the sensor never seems to pick up new issues in my st2-docker test environment. Stepping through the same code that the sensor is running it appears this is due to the default jql which is basically `project=<PROJECT_NAME>`. It seems for my JIRA project this is too many issues to pull down with no limit or pagination at once and never completes within a reasonable amount of time. 

I think we either need to make the jql configurable or change the logic in the sensor so it doesn't keep all issues (in my case over 35k) in memory. I think if we could track the latest issue ID in the project and adjust the jql to simply query issues newer than that ID, the list will be much smaller and memory footprint significantly lower for the sensor.

### PR DESCRIPTION
**Pull request by floatingstatic**
Addresses issue described in #45 

Fixes #45

### CODE REVIEW

**Comment by floatingstatic** [comment]
Hi Eugen (@armab) I wonder if I could bump this PR which addresses #45. Currently I have to use a forked version of this pack due to the size of some Jira projects in my org. The way the sensor currently works as I understand it is we load all issues in a jira project into the sensor object under `self._issues_in_project` then poll for new issues and check if that issue is in the list to determine if we should emit a trigger. With 30-40k issues this never seems to work for me due to the volume of data the jira api has to return. This change I am proposing simply loads the latest issue id with a minor modification to the jql at sensor startup and polls for issues newer than the last known id we recorded at startup. We then emit triggers for any issues newer than `self._latest_id`, and finally update `self._latest_id` after triggers are emitted. This seems more efficient and scales to very large Jira projects (and is working well for my use case with very large Jira projects). From the user's perspective this change should be a no-op (functionally the sensor produces the same result as the current logic [ie. we just diff new issues vs subsequent polls to emit triggers]).

If you are not the current maintainer of this pack perhaps you could pull in that person to help review this change? I'm also happy to help support this pack as I do for `stackstorm-napalm` if that is of any use to the st2 community.

**Review by arm4b — APPROVED** [review]
Great stuff! :100: 

JIRA is known to be used with lots of data/issues and manytimes things are slow.
This one is very helpful, thanks for the contribution!

Happy to add you as a maintainer @floatingstatic for this pack.
Could you please open a PR similar to this https://github.com/StackStorm-Exchange/stackstorm-napalm/pull/68/files in this repo to list yourself as a `jira` pack maintainer?

File: sensors/jira_sensor_for_apiv2.py [hunk 1]
```diff
@@ -73,9 +73,12 @@ def setup(self):
         self._project = self._config.get('project', None)
         if not self._project or self._project not in self._projects_available:
             raise Exception('Invalid project (%s) to track.' % self._project)
-        self._jql_query = 'project=%s' % self._project
-        all_issues = self._jira_client.search_issues(self._jql_query, maxResults=None)
-        self._issues_in_project = {issue.key: issue for issue in all_issues}
+
+        self._jql_query = 'project={} ORDER BY id DESC'.format(self._project)
```
Comment (arm4b) [inline sensors/jira_sensor_for_apiv2.py:77]: :+1:

**Review by arm4b — COMMENTED** [review]
Another addition would be storing the latest ID in the st2 K/V so if the sensor is restarted it will remember where it stopped last time.
https://docs.stackstorm.com/sensors.html#datastore-management-operations

Many pack Sensors are implemented this way, would it make sense for JIRA too? Maybe in the next PR, if you want to implement that?

**Comment by floatingstatic** [comment]
Hi @armab yeah i think using the kv store makes sense and I'm happy to add that feature in another PR (will open an issue to track this)

**Comment by floatingstatic** [comment]
Opened https://github.com/StackStorm-Exchange/stackstorm-jira/pull/51 for maintainer
Thanks for again!

**Pull request merged**